src/traffic/continuous/DimRotateCTP.cc

```diff
@@ -24,31 +24,31 @@
 
 DimRotateCTP::DimRotateCTP(
     const std::string& _name, const Component* _parent,
-    u32 _numTerminals, u32 _self, Json::Value _settings)
+    u32 _numTerminals, u32 _self, nlohmann::json _settings)
     : ContinuousTrafficPattern(_name, _parent, _numTerminals, _self,
                                _settings) {
   // parse the settings
-  assert(_settings.isMember("dimensions") &&
-         _settings["dimensions"].isArray());
-  assert(_settings.isMember("concentration") &&
-         _settings["concentration"].isUInt());
-  assert(_settings.isMember("interface_ports") &&
-         _settings["interface_ports"].isUInt());
+  assert(_settings.contains("dimensions") &&
+         _settings["dimensions"].is_array());
+  assert(_settings.contains("concentration") &&
+         _settings["concentration"].is_number_integer());
```

Comment:
Should is_number_usigned be used instead?
https://nlohmann.github.io/json/doxygen/classnlohmann_1_1basic__json_a5493f2ed1e07b0ece428bd5a47e2fb95.html#a5493f2ed1e07b0ece428bd5a47e2fb95

Reply:
I did consider this, but JSON itself does not know about signed and unsigned numbers. This JSON library stores variables as one or the other, but there is no guarantee that if you give it 16 that it will be signed or unsigned. I'll leave it as "integer" and the C++ developer should check the range.